clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -1023,9 +1023,14 @@ protected void close(Timer timer) {
     }
 
     /**
+     * Leaving the group. This method also sends LeaveGroupRequest and log {@code leaveReason} if this is dynamic members
+     * or unknown coordinator or state is not UNJOINED or this generation has a valid member id.
+     *
+     * @param leaveReason the reason to leave the group for logging
+     * @param shouldWarn should log as WARN level or INFO
      * @throws KafkaException if the rebalance callback throws exception
      */
-    public synchronized RequestFuture<Void> maybeLeaveGroup(String leaveReason) {
+    public synchronized RequestFuture<Void> maybeLeaveGroup(String leaveReason, boolean shouldWarn) throws KafkaException {
```

Comment:
`KafkaException` is a runtime exception and therefore should only be included in the javadoc. In the method signature we include checked exceptions.

Reply:
Thanks, updated.